nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,159 @@
+                    // We failed to read the blueprint. There's nothing to do
+                    // but log an error. We'll retry when we're activated again.
+                    let message = format!("{:#}", error);
+                    warn!(
+                        &log,
+                        "failed to read target blueprint";
+                        "error" => &message
+                    );
+                    json!({
+                        "error": 
+                        format!("failed to read target blueprint: {message}")})
+                }
+                (None, Ok(None)) => {
+                    // We haven't found a blueprint yet. Do nothing.
+                    json!({})
+                }
+                (Some(old), Ok(None)) => {
+                    // We have transitioned from having a blueprint to not
+                    // having one. This should not happen.
+                    let message = format!(
+                        "target blueprint with id {} was removed. There is no \
+                        longer any target blueprint",
+                        old.id
+                    );
+                    error!(&log, "{}", message);
+                    json!({"error": message})
+                }
+                (None, Ok(Some((_, new_target)))) => {
+                    // We've found a new target blueprint for the first time.
+                    // Save it and notify any watchers.
+                    let target_id = new_target.id.to_string();
+                    let time_created = new_target.time_created.to_string();
+                    info!(
+                        log,
+                        "found new target blueprint (first find)";
+                        "target_id" => &target_id,
+                        "time_created" => &time_created
+                    );
+                    self.last = Some(Arc::new(new_target));
+                    self.tx.send_replace(self.last.clone());
+                    json!({
+                        "target_id": target_id, "time_created": time_created
+                    })
+                }
+                (Some(old), Ok(Some((_, new)))) => {
+                    let target_id = new.id.to_string();
+                    let time_created = new.time_created.to_string();
+                    if old.id != new.id {
+                        // The current target blueprint has been updated
+                        info!(
+                            log,
+                            "found new target blueprint";
+                            "target_id" => &target_id,
+                            "time_created" => &time_created
+                        );
+                        self.last = Some(Arc::new(new));
+                        self.tx.send_replace(self.last.clone());
+                        json!({
+                            "target_id": target_id,
+                            "time_created": time_created
```

Comment:
Suggestion: add `time_found: Utc::now()` (so that if we're debugging some weird behavior, we can see easily when this Nexus first saw this blueprint)

Reply:
Good idea. Done in [596c9be](https://github.com/oxidecomputer/omicron/pull/4891/commits/596c9be466c0ba54c41201edf0c18d9b8c6c6280)